src/bridges/notional/NotionalBridge.sol

```diff
@@ -0,0 +1,203 @@
+            if (underlyingToken.tokenAddress != ETH) {
+                _preApprove(IERC20(underlyingToken.tokenAddress));
+            }
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /**
+     * @notice Convert function called by rollup processor, will enter or exit Notional lending position
+     * @dev Will revert if the input asset or the output asset is invalid
+     * @param _inputAssetA The input asset. For entering the market, we support underlying tokens (ETH, DAI, etc.)
+     * and their corresponding ctokens. For exiting the market, we support every valid fcash token.
+     * @param _outputAssetA The output asset. For entering the market, it should be a valid fcash token.
+     For exiting the market, this should be either an underlying token or a ctoken.
+     * @param _inputValue The amount of input token
+     * @param _interactionNonce The interaction nonce of the call
+     * @param _auxData For entering the market, this will be the market maturity. For exitng the market,
+     * this will be 0.
+     * @return outputValueA The amount of token (fcash token, ctoken, or underlying token) the bridge returned.
+     */
+    function convert(
+        AztecTypes.AztecAsset memory _inputAssetA,
+        AztecTypes.AztecAsset memory,
+        AztecTypes.AztecAsset memory _outputAssetA,
+        AztecTypes.AztecAsset memory,
+        uint256 _inputValue,
+        uint256 _interactionNonce,
+        uint64 _auxData,
+        address
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        if (_auxData != 0) {
+            outputValueA = _enter(_inputAssetA, _outputAssetA, _inputValue);
+        } else {
+            outputValueA = _exit(_inputAssetA, _outputAssetA, _inputValue);
+        }
+        if (_outputAssetA.assetType == AztecTypes.AztecAssetType.ETH) {
+            IRollupProcessor(ROLLUP_PROCESSOR).receiveEthFromBridge{value: outputValueA}(_interactionNonce);
+        }
+    }
+
+    /**
+     * @notice Helper function to compute the fcash amount
+     * @dev If the token is a ctoken, then we need to convert the amount to its underlying in 8 decimals
+     * @dev If the token is an underlying token, then we simply scale it to 8 decimals
+     * @param _token is the address of input asset token
+     * @param _underlyingToken is the address of the underlying asset of _token.
+     * @param _amount is the amount of input asset token
+     * If _token is an underlying token, then _underlyingToken == _token. If _token is a ctoken,
+     * then _underlyingToken will be _token's underlying token.
+     * @return underlyingAmount The amount of underlying token with 8 decimal
+     */
+    function computeUnderlyingAmount(
+        address _token,
+        address _underlyingToken,
+        uint256 _amount
+    ) public returns (uint88) {
```

Comment:
Why is the amount uint88 and not the default uin256?

Reply:
Notional uses `uint88` for its internal tracking.